MukurtuCMS/Mukurtu-CMS
# Color palette switcher

- Add Mukurtu Design custom module
- Add Blue Gold color palette
- Connect module to palettes to switch theme colors
- Add config page to house the palette switcher 
- Add mukurtu_site_settings module to set up a reference to the config_page on the /config page in Drupal

**To test:**
- With the branch locally, `ddev drush en mukurtu_design` (config_pages is required by mukurtu_design, so it will ask whether you want this installed as well, which you do)
- Set the default theme to Mukurtu CMS Base Theme (leave the admin theme as-is)
- Go to /admin/structure/config_pages/types and click 'Add config page', using the following settings:
- Label: Design Settings (machine name `design_settings`)
- Under Menu, menu path: /admin/config/designsettings
- Menu weight: 0
- Description: Switch the color palette of your Mukurtu installation. 
- Save
- Now go to the Manage fields tab and add a field with the following settings:
- Field type: List (text)
- Label: Design Settings Palette
- Machine name (this needs to be manually overridden to match this): `field_design_settings__palette`
- In the Allowed values, put the following:
blue-gold|Blue Gold
red-bone|Red Bone
- Set allowed number of values to 1
- Mark the field as 'Required' (this prevents an empty N/A option from showing up)
- Set the default value to Blue Gold
- Now on the Manage form display tab of the newly created design_settings content page, select "Mukurtu Palette select" as the Widget option
- Clear the cache
- Now `ddev drush en mukurtu_site_settings` 
- Now go to /admin/config and scroll to the bottom, looking for the Site Setting section
- Click Design Settings 
- Verify that the Blue and Gold palette is default
- Verify that the site palette switches when you select Red Bone and save (I've been verifying this on the "homepage" since there are links there that change colors
